hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/ChunkBufferImplWithByteBuffer.java

```diff
@@ -53,6 +54,9 @@ public boolean hasNext() {
       @Override
       public ByteBuffer next() {
         final ByteBuffer duplicated = buffer.duplicate();
+        if (!buffer.hasRemaining()) {
```

Comment:
Change LGTM. Should we move this before Line 56? Can you add a unit test for repro and verification to avoid regression in future?

Reply:
Moving to Line 56 is ok, but could you give some advice on uni test here ?

Reply:
And is rebase required here ?

Reply:
@xiaoyuyao  Please review my unit test

Reply:
Rebase is not required while GitHub says:

> This branch has no conflicts with the base branch

Even if there are conflicts, a merge from `master` is preferred over rebase, to avoid the need for force push.

Regarding unit test, I would tweak the existing one, adding to the end of https://github.com/apache/hadoop-ozone/blob/050d58b6d0f59e251519a4108930fa78f850b231/hadoop-hdds/common/src/test/java/org/apache/hadoop/ozone/common/TestChunkBuffer.java#L111-L134

these:

```
    Assert.assertFalse(i.hasNext());
    Assertions.assertThrows(NoSuchElementException.class, i::next);
```

`TestChunkBuffer` tests both implementations of `ChunkBuffer` without code duplication.

Reply:
Thanks for the detailed explanation. And how can I run checks on my own fork ?

Reply:
I think they should be run automatically when you push to your fork.  Check your repo settings at https://github.com/iamabug/hadoop-ozone/settings/actions